captum/attr/_utils/common.py

```diff
@@ -367,23 +367,68 @@ def _construct_default_feature_mask(
     return feature_mask, total_features
 
 
-def neuron_index_deprecation_decorator(func):
+def neuron_index_deprecation_decorator(func: Callable) -> Callable:
     r"""
     Decorator to deprecate neuron_index parameter for Neuron Attribution methods.
     """
 
     @functools.wraps(func)
-    def wrapper(*args, **kwargs):
+    def wrapper(*args: Any, **kwargs: Any):
         if "neuron_index" in kwargs:
             kwargs["neuron_selector"] = kwargs["neuron_index"]
             warnings.warn(
                 "neuron_index is being deprecated and replaced with neuron_selector "
-                "to support more general functionality. Please update the parameter "
+                "to support more general functionality. Please, update the parameter "
                 "name to neuron_selector. Support for neuron_index will be removed "
-                "in Captum 0.4.0",
+                "in Captum 0.4.0.",
                 DeprecationWarning,
             )
             del kwargs["neuron_index"]
         return func(*args, **kwargs)
 
     return wrapper
+
+
+def noise_tunnel_n_samples_deprecation_decorator(func: Callable) -> Callable:
+    r"""
+    Decorator to depricate n_samples parameter for NoiseTunnel's `attribute` method.
+    """
+
+    @functools.wraps(func)
+    def wrapper(*args: Any, **kwargs: Any):
+        if "n_samples" in kwargs:
```

Comment:
I think this condition can also check that nt_samples is not already in kwargs? Otherwise, it wouldn't be possible to pass both for Lime and KernelShap until this is deprecated right?

Reply:
yeah, that's a good point!